woocommerce-ac.php

```diff
@@ -1151,189 +1232,248 @@ function wcal_store_cart_timestamp() {
 						}
 					} else {
-						//$wcal_woocommerce_persistent_cart = version_compare( $woocommerce->version, '3.1.0', ">=" ) ? '_woocommerce_persistent_cart_' . get_current_blog_id() : '_woocommerce_persistent_cart' ;
-						//$updated_cart_info                = json_encode( get_user_meta( $user_id, $wcal_woocommerce_persistent_cart, true ) );
-
 						$updated_cart_info         = array();
 						$updated_cart_info['cart'] = WC()->session->cart;
 						$updated_cart_info         = json_encode( $updated_cart_info );
 
-						$query_update = "UPDATE `".$wpdb->prefix."ac_abandoned_cart_history_lite`
-										 SET abandoned_cart_info = %s,
-											 abandoned_cart_time = %d
-										 WHERE user_id      = %d
-										 AND   cart_ignored = %s ";
-						$wpdb->query( $wpdb->prepare( $query_update, $updated_cart_info, $current_time, $user_id, $cart_ignored ) );
-
-						$query_update = "SELECT * FROM `" . $wpdb->prefix . "ac_abandoned_cart_history_lite` WHERE user_id ='" . $user_id . "' AND cart_ignored='0' ";
-						$get_abandoned_record = $wpdb->get_results( $query_update );
+						$wpdb->query( //phpcs:ignore
+							$wpdb->prepare(
+								'UPDATE `' . $wpdb->prefix . 'ac_abandoned_cart_history_lite` SET abandoned_cart_info = %s, abandoned_cart_time = %d WHERE user_id      = %d AND   cart_ignored = %s',
+								$updated_cart_info,
+								$current_time,
+								$user_id,
+								$cart_ignored
+							)
+						);
+
+						$get_abandoned_record = $wpdb->get_results( //phpcs:ignore
+							$wpdb->prepare(
+								'SELECT * FROM `' . $wpdb->prefix . 'ac_abandoned_cart_history_lite` WHERE user_id = %d AND cart_ignored = %d',
+								$user_id,
+								0
+							)
+						);
 
 						if ( count( $get_abandoned_record ) > 0 ) {
-							$abandoned_cart_id   = $get_abandoned_record[0]->id;
+							$abandoned_cart_id = $get_abandoned_record[0]->id;
 							wcal_common::wcal_set_cart_session( 'abandoned_cart_id_lite', $abandoned_cart_id );
 						}
 					}
 				}
 			} else {
-				//start here guest user
+				// start here guest user.
 				$user_id = wcal_common::wcal_get_cart_session( 'user_id' );
 
 				// GDPR consent.
-				$gdpr_consent = true;
+				$gdpr_consent  = true;
 				$show_gdpr_msg = wcal_common::wcal_get_cart_session( 'wcal_cart_tracking_refused' );
 				if ( isset( $show_gdpr_msg ) && 'yes' === $show_gdpr_msg ) {
 					$gdpr_consent = false;
 				}
 
-				if ( $gdpr_consent ) {
-					$query   = "SELECT * FROM `".$wpdb->prefix."ac_abandoned_cart_history_lite` WHERE user_id = %d AND cart_ignored = '0' AND recovered_cart = '0' AND user_id != '0'";
-					$results = $wpdb->get_results( $wpdb->prepare( $query, $user_id ) );
+				if ( $gdpr_consent && $user_id > 0 ) {
+					$results = $wpdb->get_results( //phpcs:ignore
+						$wpdb->prepare(
+							'SELECT * FROM `' . $wpdb->prefix . 'ac_abandoned_cart_history_lite` WHERE user_id = %d AND cart_ignored = %d AND recovered_cart = %d',
+							$user_id,
+							0,
+							0,
```

Comment:
extra comma at the end of the line causing syntax error

Reply:
Fixed.